cmd/api/src/api/v2/database_management.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package v2
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/specterops/bloodhound/dawgs/graph"
+	"github.com/specterops/bloodhound/dawgs/ops"
+	"github.com/specterops/bloodhound/src/api"
+	"github.com/specterops/bloodhound/src/model"
+)
+
+type DatabaseManagement struct {
+	CollectedGraphData bool `json:"collectedGraphData"`
+	HighValueSelectors bool `json:"highValueSelectors"`
+	FileIngestHistory  bool `json:"fileIngestHistory"`
+	DataQualityHistory bool `json:"dataQualityHistory"`
+	AssetGroupId       int  `json:"assetGroupId"`
+}
+
+func (s Resources) HandleDatabaseManagement(response http.ResponseWriter, request *http.Request) {
+
+	var (
+		payload DatabaseManagement
+		nodeIDs []graph.ID
+		options []string
+	)
+
+	if err := api.ReadJSONRequestPayloadLimited(&payload, request); err != nil {
+		api.WriteErrorResponse(
+			request.Context(),
+			api.BuildErrorResponse(http.StatusBadRequest, "JSON malformed.", request),
+			response,
+		)
+	}
+
+	// delete graph
+	if payload.CollectedGraphData {
+		options = append(options, "collected graph data")
+
+		if err := s.Graph.ReadTransaction(request.Context(), func(tx graph.Transaction) error {
+			fetchedNodeIDs, err := ops.FetchNodeIDs(tx.Nodes())
+
+			nodeIDs = append(nodeIDs, fetchedNodeIDs...)
+			return err
+		}); err != nil {
+			api.WriteErrorResponse(
+				request.Context(),
+				api.BuildErrorResponse(http.StatusInternalServerError, fmt.Sprintf("%s: %s", "error fetching all nodes", err.Error()), request),
+				response,
+			)
+			return
+		}
+
+		if err := s.Graph.BatchOperation(request.Context(), func(batch graph.Batch) error {
+			for _, nodeId := range nodeIDs {
+				if err := batch.DeleteNode(nodeId); err != nil {
+					return err
+				}
+			}
+			return nil
+		}); err != nil {
+			api.WriteErrorResponse(
+				request.Context(),
+				api.BuildErrorResponse(http.StatusInternalServerError, fmt.Sprintf("%s: %s", "error deleting all nodes", err.Error()), request),
+				response,
+			)
+			return
+		}
+
+	}
+
+	// delete custom high value selectors
+	if payload.HighValueSelectors {
```

Comment:
What is the expectation if both `payload.CollectedGraphData` and `payload.HighValueSelectors` are true but an error occurs inside this block? Do we end up with only some data being deleted instead of all requested data deleted?

Reply:
Also, it looks like the custom selector is deleted but the nodes themselves remained tagged as part of the asset group. Is that expected behavior @StephenHinck ?

Reply:
to your first question, if a user requests to delet emultiple types of data, and gets a partial failure, maybe we should surface that to the UI?

so if i delete graph data and file ingest history, and file ingest history fails, the user would see on the UI something like "we encountered an error deleting file ingest history. please try again."